### PR TITLE
fix(shadows): background theming issue

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): closes: #707

## PR Type
What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
ShadowContainer currently strips its child's background and use it as the background for the image-brush. This is done because the inner shadows need to be drawn on top of the content background. This approach has a few problems where it doesnt subscribes to updates to the background property, nor theme updates from the `{ThemeResource}` brush.

## What is the new behavior?
ShadowContainer now creates two layers of shadows, one for background shadows and one for foreground shadows, that sandwiches the content.

## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Tested the changes where applicable:
	- [x] UWP
	- [x] WinUI
	- [x] iOS
	- [x] Android
	- [x] WASM
	- [ ] MacOS
- [x] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs [hunk 1]
```diff
@@ -19,12 +23,28 @@
 using SkiaSharp.Views.UWP;
 #endif
 
+#if __ANDROID__ && ANDROID_REFERENTIAL_IMPL
+using _SKXamlCanvas = SkiaSharp.Views.Windows.SKSwapChainPanel;
+using _SKPaintSurfaceEventArgs = SkiaSharp.Views.Windows.SKPaintGLSurfaceEventArgs;
+#else
+using _SKXamlCanvas = SkiaSharp.Views.Windows.SKXamlCanvas;
+using _SKPaintSurfaceEventArgs = SkiaSharp.Views.Windows.SKPaintSurfaceEventArgs;
+using System.Globalization;
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs:32]: move sys.glob out

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs [hunk 2]
```diff
@@ -19,12 +23,28 @@
 using SkiaSharp.Views.UWP;
 #endif
 
+#if __ANDROID__ && ANDROID_REFERENTIAL_IMPL
+using _SKXamlCanvas = SkiaSharp.Views.Windows.SKSwapChainPanel;
+using _SKPaintSurfaceEventArgs = SkiaSharp.Views.Windows.SKPaintGLSurfaceEventArgs;
+#else
+using _SKXamlCanvas = SkiaSharp.Views.Windows.SKXamlCanvas;
+using _SKPaintSurfaceEventArgs = SkiaSharp.Views.Windows.SKPaintSurfaceEventArgs;
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs:31]: add a permutation for `SkiaSharp.Views.Windows.*` aliases where `Windows` can be: Windows or UWP (based on `#if IS_WINUI`)

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs [hunk 3]
```diff
@@ -19,12 +23,28 @@
...
+using System.Globalization;
+
+#endif
+
 namespace Uno.Toolkit.UI;
 
+// fixme@xy: to validate
+// it seems like 1st/2nd original author favored float
+// we did switch it to double, maybe it is require by certain platform skia impl?
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs:40]: remove // solved

File: src/Uno.Toolkit.UI/Extensions/DependencyObjectExtensions.cs
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Extensions/DependencyObjectExtensions.cs:1]: debug fix, kinda out of scope
remove?

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs (hunk 2 above)
Comment (Xiaoy312) [reply src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs]: nvm... there is only winui build for this

**Comment by Xiaoy312** [comment]
@agneszitte im unlinking 707, because there are currently multiple issues in that one
and this only addresses one of them
(i explicitly used `re` instead of `closes` keyword)

**Comment by Xiaoy312** [comment]
sanity-check done: winui, skia, wasm, ios, droid

**Review by kazo0 — COMMENTED** [review]
I'm seeing some strange behaviors if I add an extra shadow using the ShadowContainerSamplePage here:

Before adding new shadow:
image

After adding new shadow:
image

It seems to also break on `main` but in a different way: 

Before adding new shadow:
image

After adding new shadow:

image


Not sure if it's really related to these changes but something is different

**Comment by Xiaoy312** [comment]
@kazo0 There might be re-layout issue with the SC on skia, that was previously not much noticeable due to background/border breaking on changes on skia. I think this just uncovers it.

**Review by kazo0 — APPROVED** [review]

**Review by pictos — APPROVED** [review]

**Comment by agneszitte** [comment]
@Mergifyio backport legacy/3x

**Pull request merged**